src/value/from.rs

```diff
@@ -9,7 +10,7 @@ macro_rules! from_integer {
         $(
             impl From<$ty> for Value {
                 fn from(n: $ty) -> Self {
-                    Value::I64(n as i64)
+                    Value::Number(Number::from(n as i64))
```

Comment:
IIUC, you can omit the `as i64` on this line as `Number` already implements from for a bunch of integer sizes and just change the call below to

```rust
from_integer! {
    i8 i16 i32 i64 isize
    u8 u16 u32 u64 usize
}
```